certification/engine/engine.go

```diff
@@ -28,137 +28,107 @@ type CheckEngine interface {
-		// refuse to run if the user has not specified any checks
-		return nil, fmt.Errorf("no checks have been enabled")
-	}
-
-	checks := make([]certification.Check, 0, len(config.EnabledChecks))
-	for _, checkString := range config.EnabledChecks {
-		check := queryChecks(checkString)
-		if check == nil {
-			err := fmt.Errorf("requested check not found: %s", checkString)
-			return nil, err
-		}
-
-		checks = append(checks, check)
+func NewForConfig(ctx context.Context, cfg certification.Config) (CheckEngine, error) {
+	checks, err := initializeChecks(ctx, cfg.Policy(), cfg)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing checks: %v", err)
 	}
 
 	engine := &internal.CraneEngine{
-		Image:     config.Image,
+		Config:    cfg,
+		Image:     cfg.Image(),
 		Checks:    checks,
-		IsBundle:  config.Bundle,
-		IsScratch: config.Scratch,
+		IsBundle:  cfg.IsBundle(),
+		IsScratch: cfg.IsScratch(),
 	}
 
 	return engine, nil
 }
 
-// queryChecks queries Operator and Container checks by name, and return certification.Check
-// if found; nil otherwise. This will be collapsed when old checks are all deprecated.
-func queryChecks(checkName string) certification.Check {
-	// query Operator checks
-	if check, exists := operatorPolicy[checkName]; exists {
-		return check
-	}
-	// if not found in Operator Policy, query container policy.
-	// No need to check scratch container policy since this is
-	// a superset.
-	if check, exists := containerPolicy[checkName]; exists {
-		return check
+// initializeChecks configures checks for a given policy p using cfg as needed.
+func initializeChecks(ctx context.Context, p policy.Policy, cfg certification.Config) ([]certification.Check, error) {
+	switch p {
+	case policy.PolicyOperator:
+		return []certification.Check{
+			operatorpol.NewScorecardBasicSpecCheck(internal.NewOperatorSdkEngine(cfg.ScorecardImage()), cfg.Namespace(), cfg.ServiceAccount(), cfg.Kubeconfig(), cfg.ScorecardWaitTime()),
+			operatorpol.NewScorecardOlmSuiteCheck(internal.NewOperatorSdkEngine(cfg.ScorecardImage()), cfg.Namespace(), cfg.ServiceAccount(), cfg.Kubeconfig(), cfg.ScorecardWaitTime()),
+			operatorpol.NewDeployableByOlmCheck(internal.NewOperatorSdkEngine(cfg.ScorecardImage()), cfg.IndexImage(), cfg.DockerConfig(), cfg.Channel()),
+			operatorpol.NewValidateOperatorBundleCheck(internal.NewOperatorSdkEngine(cfg.ScorecardImage())),
+		}, nil
+	case policy.PolicyContainer:
+		return []certification.Check{
+			&containerpol.HasLicenseCheck{},
+			containerpol.NewHasUniqueTagCheck(cfg.DockerConfig()),
+			&containerpol.MaxLayersCheck{},
+			&containerpol.HasNoProhibitedPackagesCheck{},
+			&containerpol.HasRequiredLabelsCheck{},
+			&containerpol.RunAsNonRootCheck{},
+			&containerpol.HasModifiedFilesCheck{},
+			containerpol.NewBasedOnUbiCheck(pyxis.NewPyxisClient(
+				certification.DefaultPyxisHost,
+				cfg.PyxisAPIToken(),
+				cfg.CertificationProjectID(),
+				&http.Client{Timeout: 60 * time.Second})),
```

Comment:
Since we have a context, it should probably be used here, but that can be a later PR.